### PR TITLE
Feature/timeseries group

### PR DESCRIPTION
**Pull request by adamkorynta**

### CODE REVIEW

File: cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import java.util.Objects;
+import javax.validation.Valid;
+
+/**
+ * A representation of a timeseries group
+ */
+@ApiModel(description = "A representation of a timeseries group")
+@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2021-11-02T12:25:34.578-07:00[America/Los_Angeles]")
```
Comment (rma-psmorris) [inline cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java:36]: no office id?

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
Where is the time series category model class?

File: cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java:36]: Hmm the office id is part of the category. There is a separate column in the AV_TS_CAT_GRP view so I'll have to request that they return it to us.

**Comment by adamkorynta** [comment]
https://github.com/HydrologicEngineeringCenter/cwms-radar-client/pull/12#pullrequestreview-810508794

The TimeSeriesCategory model class didn't receive any updates in this PR. That was part of  PR 11: https://github.com/HydrologicEngineeringCenter/cwms-radar-client/pull/11/files#diff-3dc3610338c91ab786e64cb4b27ec1faf7e12fd4325b2d76a00498df5286c34b

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesGroupController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import mil.army.usace.hec.cwms.radar.client.model.RadarObjectMapper;
+import mil.army.usace.hec.cwms.radar.client.model.TimeSeriesGroup;
+
+public final class TimeSeriesGroupController {
+
+    private static final String TIME_SERIES_GROUP_ENDPOINT = "timeseries/group";
+
+    public TimeSeriesGroup retrieveTimeSeriesGroup(ApiConnectionInfo apiConnectionInfo, TimeSeriesGroupEndpointInput timeSeriesEndpointInput)
+        throws IOException {
+        String endpoint = timeSeriesEndpointInput.getGroupId()
```
[the remark this answers is not in the document]
Comment (adamkorynta) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesGroupController.java:41]: For existing API's that would be ok, but the REST API design allows for parameterization at any level. So you can have http://localhost:1111/timeseries/<parameter>/metadata

I'd also argue that building up the endpoint is the single responsibility of the controller object

File: cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/TimeSeriesGroup.java:36]: Created a ticket: https://github.com/USACE/cwms-radar-api/issues/109

**Review by rma-psmorris — APPROVED** [review]
approve

**Pull request merged**